### PR TITLE
Fix transaction issue in ingest flow

### PR DESCRIPTION
**Pull request by rmn-boiko**
## Description

Closes: #issue

## Checklist before requesting a review

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅ 
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [x] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [x] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅

### CODE REVIEW

File: src/domain/datasets/domain/src/services/dataset_key_value_service.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatasetKeyValueService: Sync + Send {
+    async fn get_dataset_env_var_value_by_key<'a>(
+        &self,
+        dataset_env_var_key: &str,
+        dataset_env_vars: &'a [DatasetEnvVar],
+    ) -> Result<Secret<String>, GetDatasetEnvVarError>;
```
Comment (zaychenko-sergei) [inline src/domain/datasets/domain/src/services/dataset_key_value_service.rs:22]: Not all of those are secrets, so maybe an enum in result could be a little nicer.

Comment (zaychenko-sergei) [inline src/domain/datasets/domain/src/services/dataset_key_value_service.rs:22]: As you renamed service, change the method name and error name too.

File: src/domain/datasets/services/src/dataset_key_value_service_impl.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#[async_trait::async_trait]
+impl DatasetKeyValueService for DatasetKeyValueServiceImpl {
+    async fn get_dataset_env_var_value_by_key<'a>(
+        &self,
+        dataset_env_var_key: &str,
+        dataset_env_vars: &'a [DatasetEnvVar],
+    ) -> Result<Secret<String>, GetDatasetEnvVarError> {
+        if let Some(existing_dataset_env_var) = dataset_env_vars
+            .iter()
+            .find(|dataset_env_var| dataset_env_var.key == dataset_env_var_key)
```
Comment (zaychenko-sergei) [inline src/domain/datasets/services/src/dataset_key_value_service_impl.rs:59]: Hash map plz

File: src/domain/task-system/services/src/task_executor_impl.rs [hunk 3]
```diff
@@ -99,9 +102,34 @@ impl TaskExecutor for TaskExecutorImpl {
 
             let outcome = match &task.logical_plan {
                 LogicalPlan::UpdateDataset(upd) => {
+                    let dataset_env_vars = DatabaseTransactionRunner::new(self.catalog.clone())
+                        .transactional_with(
+                            |dataset_env_vars_svc: Arc<dyn DatasetEnvVarService>| async move {
+                                let dataset_env_vars = dataset_env_vars_svc
+                                    .get_all_dataset_env_vars_by_dataset_id(
```
Comment (zaychenko-sergei) [inline src/domain/task-system/services/src/task_executor_impl.rs:109]: This code is growing too big, let's split handlers for each type of logical plan into separate methods for the beginning.

File: src/domain/datasets/domain/src/services/dataset_key_value_service.rs (hunk 1 above)
Comment (rmn-boiko) [reply src/domain/datasets/domain/src/services/dataset_key_value_service.rs:22]: Just a small note. In static implementation where we fetch from env vars we cannot find is it secret or not so I put here as always secret value

File: src/domain/task-system/services/Cargo.toml [hunk 4]
```diff
@@ -22,9 +22,11 @@ doctest = false
 
 
 [dependencies]
+database-common = { workspace = true }
 event-bus = { workspace = true }
 opendatafabric = { workspace = true }
 kamu-core = { workspace = true }
+kamu-datasets-services = { workspace = true }
```
Comment (zaychenko-sergei) [inline src/domain/task-system/services/Cargo.toml:29]: Why do we have to link to service implementation crate and not the domain crate?

Comment (zaychenko-sergei) [reply src/domain/task-system/services/Cargo.toml:29]: I only see imports like 

`use kamu_datasets_services::domain::{DatasetEnvVar, DatasetEnvVarService};`

Comment (rmn-boiko) [reply src/domain/task-system/services/Cargo.toml:29]: great catch. Thanks for pointing. Probably VScode autoimport did that and I missed to doublecheck. FIxed and double checked

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**